private/bufpkg/bufconfig/bufconfig.go

```diff
@@ -102,6 +102,16 @@ func WriteConfig(
 // WriteConfigOption is an option for WriteConfig.
 type WriteConfigOption func(*writeConfigOptions)
 
+func WriteConfigWithName(name string) (WriteConfigOption, error) {
```

Comment:
Does this follow the style guide?

Reply:
I think so. This conforms with the code surrounded. The parameter name is the module name passed in by user which has a key called ```name``` in the config file.

Reply:
I was trying to hide `moduleIdentity` in ```modinit.go``` since ´modinit.go´ is only setting the config with given parameters. 
I also agree that pulling the `bufmoduleref.ModuleIdentityForString(name)` call to `modinit.go` will make the functions uniform to other wrappers.

Reply:
All of our public types need documentation - looking at the style guide, this doesn't appear to be in there, so we should add

Reply:
The way of setting this config is changed. This function is no longer needed. 

I'll take a look at the style guide and keep in mind in future.